src/tsickle.ts

```diff
@@ -634,6 +634,11 @@ class Annotator extends ClosureRewriter {
    *     emit it as is and visit its children.
    */
   maybeProcess(node: ts.Node): boolean {
+    if (this.file.statements.indexOf(node as ts.Statement) !== -1) {
```

Comment:
this is O(n) on each statement, so O(n^2) in the number of statements in the source file.